src/isomorphic/classic/element/__tests__/ReactElement-test.js

```diff
@@ -58,6 +58,66 @@ describe('ReactElement', function() {
     expect(element.props).toEqual(expectation);
   });
 
+  it('should warn when `key` is being accessed', function() {
+    spyOn(console, 'error');
+    var container = document.createElement('div');
+    var Child =  React.createClass({
+      render: function() {
+        return <div> {this.props.key} </div>
+      },
+    });
+    var Parent = React.createClass({
+      render: function() {
+        return (
+          <div>
+            <Child key="0" />
+            <Child key="1" />
+            <Child key="2" />
+          </div>
+        );
+      },
+    });
+    expect(console.error.calls.length).toBe(0);
+    ReactDOM.render(<Parent />, container);
+    expect(console.error.calls.length).toBe(3);
```

Comment:
Check a file like `ReactElementValidator` how a module-global object (`ownerHasKeyUseWarning`) is used to make sure the warning is printed only once for each offense.